### DISCUSSION TITLE
CharacterVirtual doesn't interact with scene queries (Shape Casts, Ray Casts, etc...)

### DISCUSSION THREAD

**Discussion by peter1745** [jrouwe/JoltPhysics#543] [Q&A] [answered]
Hey! Got a quick question, are there any plans on extending the typical scene queries to detect hits on CharacterVirtual instances? During the Ludum Dare game jam I noticed that Jolt doesn't seem to detect when a shape cast or ray cast hits a shape belonging to a CharacterVirtual.
It would be very useful to be able to detect that, but from what I could tell it's because Characters aren't backed by a physics body, and the scene queries only operate on Bodies.

Are there any plans on adding native support for this? I know that I could probably work around it by explicitly creating a Body for any Character that needs to be detectable by scene queries, and ensuring that the backing body is always in the same location as the character, but I'd rather avoid it unless I have to

**Comment by jrouwe** [comment]
CharacterVirtuals are indeed not backed by a physics body, but you can easily create a TransformedShape from the character's (center of mass) position and shape and use that to do shape/ray casts against (just call that in addition to your regular query on the scene).

Please see #239 for more information on how you could make an internal rigid body, which is indeed the alternative way of fixing this.

**Reply by peter1745** [reply]
Hmm not quite sure I fully understand what you mean with creating a TransformedShape, do you mean adding a TransformedShape to a normal body and checking for queries against that? Or am I not understanding it fully?

Thanks for the reference by the way, I'll be sure to take a look at it today!

**Reply by jrouwe** [reply, marked as answer]
The TransformedShape class can be constructed from the position and shape of the character and offers a CastRay etc. function. If you pass the same collision collector you're passing to the scene to it you'll get a combined collision result for scene + character

**Reply by peter1745** [reply]
Ahh I see, I'll have to look into that then, thank you for the help!
